### PR TITLE
Tabular tasks

### PR DESCRIPTION
**Pull request by merveenoyan**
This PR..
- Adds skops and Hub+Gradio integration to `tabular-classification` & `tabular-regression`
- Adds metrics to `tabular-regression`
- Adds various sections to `tabular-regression` & `tabular-classification` (dataset, widget models and more)
- Puts `tabular-regression` to live 🙂

### CODE REVIEW

**Comment by merveenoyan** [comment]
Putting them here so you don't have to get a build.
Ekran Resmi 2022-12-16 19 12 48

Ekran Resmi 2022-12-16 19 12 06
Ekran Resmi 2022-12-16 19 12 22
Ekran Resmi 2022-12-16 19 12 32
Ekran Resmi 2022-12-16 19 12 42

File: tasks/src/tabular-classification/about.md [hunk 1]
```diff
@@ -15,18 +15,40 @@ Tabular classification models can be used in detecting fraudulent credit card tr
...
 ```python
 from skops import hub_utils
-import joblib
+# initialize a repository with a trained model
+hub_utils.init(model, dst=local_repo)
+# push to Hub!
+hub_utils.push("username/my-awesome-model", source=local_repo)
+```
+
+Once model is pushed, you can infer easily.
```
Comment (BenjaminBossan) [inline tasks/src/tabular-classification/about.md:37]:
```suggestion
Once the model is pushed, you can infer easily.
```

File: tasks/src/tabular-classification/about.md [hunk 2]
```diff
@@ -15,18 +15,40 @@ Tabular classification models can be used in detecting fraudulent credit card tr
...
+- Programmatically create model cards,
+- Securely serialize your scikit-learn model. (See limitations of using pickle [here](https://huggingface.co/docs/hub/security-pickle).)
+
+You can push your model as follows: 
 
 ```python
 from skops import hub_utils
-import joblib
+# initialize a repository with a trained model
+hub_utils.init(model, dst=local_repo)
```
Comment (BenjaminBossan) [inline tasks/src/tabular-classification/about.md:32]: Do you think we should explain what "local_repo" is?

File: tasks/src/tabular-regression/about.md [hunk 3]
```diff
@@ -36,8 +28,47 @@ model.fit(X, y)
...
+
+```python
+from skops import hub_utils
+# initialize a repository with a trained model
+hub_utils.init(model, dst=local_repo)
+# push to Hub!
+hub_utils.push("username/my-awesome-model", source=local_repo)
+```
+
+Once model is pushed, you can infer easily.
```
Comment (BenjaminBossan) [inline tasks/src/tabular-regression/about.md:49]:
```suggestion
Once the model is pushed, you can infer easily.
```

**Review by BenjaminBossan — COMMENTED** [review]
Overall, looks good to me, I have a few comments, please take a look.

Some more general points:

There are some duplications between the two tasks, is it possible with the hub docs to put them in a separate file and load them in? Otherwise, we would always have to pay attention to keep both in sync.

Not from your PR, but maybe worth fixing:

In the `about.md` for classification and regression, there are possibly some issues, see here:

- https://github.com/huggingface/hub-docs/blob/eae90299595f4f9d1d78570927b6e23e5c9190d7/tasks/src/tabular-classification/about.md
- https://github.com/huggingface/hub-docs/blob/eae90299595f4f9d1d78570927b6e23e5c9190d7/tasks/src/tabular-regression/about.md

E.g. "### Fraud Detection" is not correctly recognized as a section, at least not in the GH preview. Possible reasons: need an empty line after the title, or the space after `###` is not an actual space but some invisible char. This issue comes up a few times.

Additionally, in the [`about.md` for regression](https://raw.githubusercontent.com/huggingface/hub-docs/eae90299595f4f9d1d78570927b6e23e5c9190d7/tasks/src/tabular-regression/about.md), there are two subsections for "Use Cases", but they use the same amount of `#` -- there should be 3, not 2.

**Comment by merveenoyan** [comment]
@BenjaminBossan if you could see it in screenshots Fraud Detection is recognized as section. I don't think it's worth refactoring task pages as they aren't code and it's a crowd-sourced documentation effort so it's also confusing for the contributors. I don't see two Use Cases in one page, can you point it out?

**Comment by BenjaminBossan** [comment]
> if you could see it in screenshots Fraud Detection is recognized as section.

Yes, I saw that, probably GH interprets markdown differently, not sure what the "more correct" interpretation is.

> I don't think it's worth refactoring task pages as they aren't code and it's a crowd-sourced documentation effort so it's also confusing for the contributors

Okay, makes sense

> I don't see two Use Cases in one page, can you point it out?

For regression, should the two examples under "Use Cases" not be subsections of it?:

```markdown
## Use Cases

- ## Sales Prediction: a Use Case for Predicting a Continuous Target Variable 
+ ### Sales Prediction: a Use Case for Predicting a Continuous Target Variable 

...

- ## Filling Missing Values
+ ### Filling Missing Values

...
```

Regarding the "Filling Missing Values", I'm not really sure if that's a "use case", but at least "Sales Prediction" looks like it should be.

**Comment by merveenoyan** [comment]
@BenjaminBossan ah I understand you know! thanks, I will align them! 🙂

File: tasks/src/tabular-classification/about.md [hunk 4]
```diff
@@ -15,18 +15,41 @@ Tabular classification models can be used in detecting fraudulent credit card tr
 ### Churn Prediction
 Tabular classification models can be used in predicting customer churn in telecommunication. An example dataset for the task is hosted [here](https://huggingface.co/datasets/scikit-learn/churn-prediction).
 
-## Model Hosting and Inference
 
-You can use the [skops](https://skops.readthedocs.io/en/stable/) library to share, explore, and use `scikit-learn` models on the Hugging Face Hub. `skops` models have widgets to try the models on the browser and have descriptive reports (also known as model cards) in their repositories. You can pull a `scikit-learn` model like below using `skops`:
+# Model Hosting and Inference 
 
+You can use [skops](https://skops.readthedocs.io/) for model hosting and inference on 🤗Hub. Using `skops` you can:
```
Comment (osanseviero) [inline tasks/src/tabular-classification/about.md:21]:
```suggestion
You can use [skops](https://skops.readthedocs.io/) for model hosting and inference on the Hugging Face Hub. Using `skops` you can:
```

File: tasks/src/tabular-classification/about.md [hunk 5]
```diff
@@ -15,18 +15,41 @@ Tabular classification models can be used in detecting fraudulent credit card tr
...
+- Easily use inference API,
+- Build Gradio UIs with one line of code,
```
Comment (osanseviero) [inline tasks/src/tabular-classification/about.md:23]: Not sure our audience would know what Gradio is at this point. Maybe say "Build demos with one line of code"

File: tasks/src/tabular-classification/about.md [hunk 6]
```diff
@@ -15,18 +15,41 @@ Tabular classification models can be used in detecting fraudulent credit card tr
...
+- Programmatically create model cards,
+- Securely serialize your scikit-learn model. (See limitations of using pickle [here](https://huggingface.co/docs/hub/security-pickle).)
```
Comment (osanseviero) [inline tasks/src/tabular-classification/about.md:25]: Not sure all the audience will know what scikit-learn is :thinking:

File: tasks/src/tabular-classification/about.md [hunk 7]
```diff
@@ -15,18 +15,41 @@ Tabular classification models can be used in detecting fraudulent credit card tr
...
+```
 
-hub_utils.download(repo_id="user-name/my-awesome-model", dst=target_path)
-model = joblib.load(Path(target_path)/"model.pkl")
-model.predict(sample)
+You can launch a UI for your model with only one line of code!
+
+```python
+import gradio as gr
+gr.Interface.load("huggingface/username/my-awesome-model").launch()
```
Comment (osanseviero) [inline tasks/src/tabular-classification/about.md:52]: Very cool! :fire:

File: tasks/src/tabular-classification/data.ts [hunk 8]
```diff
@@ -6,12 +6,8 @@ import { TASKS_MODEL_LIBRARIES } from "../const";
 const taskData: TaskData = {
 	datasets: [
 		{
-			description: "Binary classification dataset based on a census on income.",
-			id:          "scikit-learn/adult-census-income",
-		},
-		{
-			description: "Multi-class dataset on iris flower species.",
-			id:          "scikit-learn/iris",
+			description: "A comprehensive curation of datasets covering all benchmarks.",
```
Comment (osanseviero) [inline tasks/src/tabular-classification/data.ts:9]: Should we bring one one of the above? That way we'll have two datasets

File: tasks/src/tabular-regression/about.md [hunk 9]
```diff
@@ -36,8 +28,48 @@ model.fit(X, y)
 ```
 
 
+# Model Hosting and Inference
+
+You can use [skops](https://skops.readthedocs.io/) for model hosting and inference on 🤗Hub. Using `skops` you can:
+- Easily use inference API,
+- Build Gradio UIs with one line of code,
+- Programmatically create model cards,
+- Securely serialize your scikit-learn model. (See limitations of using pickle [here](https://huggingface.co/docs/hub/security-pickle).)
```
Comment (osanseviero) [inline tasks/src/tabular-regression/about.md:37]: Same comments as for tabular classification

File: tasks/src/tabular-regression/about.md [hunk 10]
```diff
@@ -36,8 +28,48 @@ model.fit(X, y)
 ```
 
 
+# Model Hosting and Inference
+
+You can use [skops](https://skops.readthedocs.io/) for model hosting and inference on 🤗Hub. Using `skops` you can:
```
Comment (osanseviero) [inline tasks/src/tabular-regression/about.md:33]:
```suggestion
You can use [skops](https://skops.readthedocs.io/) for model hosting and inference on the Hugging Face Hub. Using `skops` you can:
```

File: tasks/src/tabular-regression/data.ts [hunk 11]
```diff
@@ -6,13 +6,9 @@ import { TASKS_MODEL_LIBRARIES } from "../const";
 const taskData: TaskData = {
 	datasets: [
 		{
-			description: "Dataset containing tip amounts for food servers.",
-			id:          "scikit-learn/tips",
+			description: "A comprehensive curation of datasets covering all benchmarks.",
```
Comment (osanseviero) [inline tasks/src/tabular-regression/data.ts:9]: Should we keep one of the others still, even if overlapping?

**Review by osanseviero — APPROVED** [review]
Nice, thanks a lot! :fire:

File: tasks/src/tabular-classification/data.ts (hunk 8 above)
Comment (merveenoyan) [reply tasks/src/tabular-classification/data.ts:9]: the benchmark already has those datasets.

File: tasks/src/tabular-regression/about.md (hunk 9 above)
Comment (merveenoyan) [reply tasks/src/tabular-regression/about.md:37]: the benchmark already has those datasets.

**Review by BenjaminBossan — APPROVED** [review]
Great addition, LGTM, thanks.

File: tasks/src/tabular-classification/about.md (hunk 6 above)
Comment (merveenoyan) [reply tasks/src/tabular-classification/about.md:25]: I will put a note but it really is in 101 of these tasks.

**Comment by merveenoyan** [comment]
@osanseviero I applied your comments and solved merge conflict.

File: tasks/src/tabular-classification/about.md (hunk 6 above)
Comment (merveenoyan) [reply tasks/src/tabular-classification/about.md:25]: addressed this.

File: tasks/src/tabular-classification/about.md (hunk 5 above)
Comment (merveenoyan) [reply tasks/src/tabular-classification/about.md]: Addressed this.

**Comment by osanseviero** [comment]
Looks great! Thanks a lot :fire:

**Pull request merged**